packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -2367,8 +2371,25 @@ class ComboBoxInternal extends React.Component<IComboBoxInternalProps, IComboBox
    */
   private _getAriaActiveDescendantValue(): string | undefined {
     const { currentOptions, selectedIndices } = this.props.hoisted;
-    const { isOpen, currentPendingValueValidIndex } = this.state;
+    const { isOpen, currentPendingValue } = this.state;
     const options = currentOptions.map((item, index) => ({ ...item, index }));
+    let currentPendingValueValidIndex = this.state.currentPendingValueValidIndex;
+
+    // We should check if the pending value index is withing the bounds of the options and that it matches the text of
+    // the pending value. If it does not match, we should update it the same way we do in _onInputChange. This is
+    // needed since the user may be filtering the options and the pending value and index are only updated on input
+    // change.
+    if (
+      currentPendingValue &&
+      (currentPendingValueValidIndex >= options.length ||
+        options[currentPendingValueValidIndex].text.indexOf(currentPendingValue) < 0)
+    ) {
+      currentPendingValueValidIndex =
+        this.props.allowFreeform || this.props.allowFreeInput
+          ? this._processInputChangeWithFreeform(currentPendingValue)
+          : this._processInputChangeWithoutFreeform(currentPendingValue);
```

Comment:
Having these functions here is an issue, since we are updating the state in the render function. One possible solution could be to add `aria-activedescendant` to the state and update its value in `componentDidUpdate` that way we could also update based on the options changing.

@smhigley and @khmakoto what do you think?